common/src/main/java/net/opentsdb/query/MetaQuery.java

```diff
@@ -0,0 +1,270 @@
+import com.google.common.base.Strings;
+import net.opentsdb.core.TSDB;
+import net.opentsdb.query.filter.QueryFilter;
+import net.opentsdb.query.filter.QueryFilterFactory;
+
+/**
+ * Represents parameters to search for metadata.
+ *
+ * @since 3.0
+ */
+@SuppressWarnings("serial")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class MetaQuery {
+
+  /**
+   * The starting pointer for results for pagination
+   */
+  private String from;
+
+  /**
+   * The end pointer for results for pagination
+   */
+  private String to;
+
+  /**
+   * The namespace for the query
+   */
+  @JsonProperty("namespace")
+  private String namespace;
+
+  /**
+   * Filters for the query. Can be a chained filter or a single filter
+   */
+  private QueryFilter filters;
+
+  /**
+   * The field by which user wants to aggregate by to return unique values
+   */
+  private Field aggregateBy = Field.ALL;
+
+  /**
+   * The tag for which a second level aggregation is applied. Lists tag values for a tag key
+   */
+  private String aggregationField;
+
+  /**
+   * Size of number unique tag values to return.
+   */
+  private String size;
+
+  protected MetaQuery(final Builder builder) {
+
+    from = Strings.isNullOrEmpty(builder.from) ? null : builder.from;
+
+    to = Strings.isNullOrEmpty(builder.to) ? null : builder.to;
+
+    namespace = Strings.isNullOrEmpty(builder.namespace) ? null : builder.namespace;
+
+    aggregationField = Strings.isNullOrEmpty(builder.aggregationField) ? null : builder.aggregationField;
+
+    if (builder.filters == null) {
+      throw new IllegalArgumentException("Please set atleast one filter");
+    } else {
+      filters = builder.filters;
+    }
+
+    aggregateBy = builder.aggregateBy == null ? Field.ALL : builder.aggregateBy;
+
+    size = Strings.isNullOrEmpty(builder.size) ? null : builder.size;
+
+    if (builder.from == null) {
+      throw new IllegalArgumentException("Please set from field");
+    }
+
+    if (builder.to == null) {
+      throw new IllegalArgumentException("Please set to field");
+    }
+
+    if (builder.namespace == null) {
+      throw new IllegalArgumentException("Please set a namespace");
+    }
+
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+
+  public String getFrom() {
+    return from;
+  }
+
+
+  public String getTo() {
+    return to;
+  }
+
+  public QueryFilter getFilters() {
+    return filters;
+  }
+
+
+  public Field getAggregateBy() {
+    return aggregateBy;
+  }
+
+  public String getAggregationField() {
+    return aggregationField;
+  }
+
+
+  public static MetaQuery.Builder newBuilder() {
+    return new Builder();
+  }
+
+
+  /**
+   * Builder through which the query is parsed and parameters are set
+   */
+  public static class Builder {
+    private String from;
+    private String to;
+    private String namespace;
+    private QueryFilter filters;
+    private Field aggregateBy;
+    private String aggregationField;
+    private String size;
```

Comment:
ints where appropriate. We can default from to 0, to to Integer.MAX_VALUE() size to 25 or something.